pkg/route/api/validation/validation.go

```diff
@@ -125,18 +125,18 @@ func ExtendedValidateRoute(route *routeapi.Route) field.ErrorList {
 	//       break, so disable the hostname validation for now.
 	// hostname := route.Spec.Host
 	hostname := ""
-	var certPool *x509.CertPool
+	var verifyOptions *x509.VerifyOptions
 
 	if len(tlsConfig.CACertificate) > 0 {
-		certPool = x509.NewCertPool()
+		certPool := x509.NewCertPool()
 		if ok := certPool.AppendCertsFromPEM([]byte(tlsConfig.CACertificate)); !ok {
 			result = append(result, field.Invalid(tlsFieldPath.Child("caCertificate"), tlsConfig.CACertificate, "failed to parse CA certificate"))
 		}
-	}
 
-	verifyOptions := &x509.VerifyOptions{
-		DNSName: hostname,
-		Roots:   certPool,
+		verifyOptions = &x509.VerifyOptions{
+			DNSName: hostname,
+			Roots:   certPool,
+		}
 	}
```

Comment:
intermediates would have to be part of the CA cert. will add the above tests and check. thx